controlplane/kubeadm/controllers/controller.go

```diff
@@ -254,6 +254,12 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Ensure kubeadm role bindings for v1.18+
+	if err := workloadCluster.AllowBoostrapTokensToGetNodes(ctx); err != nil {
```

Comment:
Do we need to do this before L224?

Reply:
The role is only used for worker nodes, but doesn't hurt.

Reply:
So not all joins but specifically only non-CP joins?

Reply:
all joining nodes that use a kubeadm 1.18+ binary needs this rbac to be set in the cluster.